lib/checks/aria/aria-allowed-attr-evaluate.js

```diff
@@ -31,13 +32,42 @@ function ariaAllowedAttrEvaluate(node, options, virtualNode) {
   const role = getRole(virtualNode);
   const attrs = virtualNode.attrNames;
   let allowed = allowedAttr(role);
-
   // @deprecated: allowed attr options to pass more attrs.
   // configure the standards spec instead
   if (Array.isArray(options[role])) {
     allowed = uniqueArray(options[role].concat(allowed));
   }
 
+  // TODO: look into memoizing getRole and removing this local cache
+  let tableMap = cache.get('aria-allowed-attr-table');
+  if (!tableMap) {
+    tableMap = new WeakMap();
+    cache.set('aria-allowed-attr-table', tableMap);
+  }
+
+  // check if the parent exists otherwise a TypeError will occur (virtual-nodes specifically)
+  if (virtualNode.parent) {
```

Comment:
We'll only want to do this parent lookup if we are on a `row` element which has one of the attributes we care about. Easiest way would be to mimic what we've done in https://github.com/dequelabs/axe-core/blob/develop/lib/checks/aria/aria-valid-attr-value-evaluate.js#L36 with our prechecks inside of the attribute name for loop on line 74. Something like:

```suggestion
  if (virtualNode.parent && role === 'row') {
```

```js
function validateRowAttrs() {
  /* code to check parent */
}

const preChecks = {
  'aria-postinset': validateRowAttrs
  'aria-setsize': validateRowAttrs,
  etc.
}

if (allowed) {
    for (let i = 0; i < attrs.length; i++) {
      const attrName = attrs[i];
      if (validateAttr(attrName) && !allowed.includes(attrName) && (preChecks[attrName] ? preChecks[attrName](validValue) : true)) {
```

Reply:
Maybe I am not understanding, but can this be done in the same if statement? As we create our own data shape: 
```JS
 this.data({
        messageKey: 'table',
        values: validValue,
        tableRole
      });
```

Reply:
After deep diving into both files, I was able to mimic similar to what you suggested. Please let me know your thoughts on the newly updated solution!